lib/configs/all.ts

```diff
@@ -0,0 +1,71 @@
+export = {
+    plugins: ["regexp"],
+    rules: {
+        // ESLint core rules
+        "no-control-regex": "error",
```

Comment:
I still think that it would be better to generate this config at runtime.

We might have to disable/configure more ESLint core rules in the future and I don't want this config to become out of date.

Reply:
I think the config of `eslint:all` will be helpful for you.
https://github.com/eslint/eslint/blob/master/conf/eslint-all.js

You can get all the rules from the following files.
https://github.com/ota-meshi/eslint-plugin-regexp/blob/master/lib/utils/rules.ts

You can access the rule name from `rule.meta.docs.ruleId`.
https://github.com/ota-meshi/eslint-plugin-regexp/blob/master/lib/types.ts#L22
https://github.com/ota-meshi/eslint-plugin-regexp/blob/master/lib/utils/index.ts#L177

Reply:
And, I was wondering how to make a this `all.ts` always include every single plugin's rule.
Maybe it should be automatically generated from `recommended.ts` with concatenating parts/lines of `lib\utils\rules.ts` ?

Because I made this `all.ts` based on `recommended.ts` :
I ran `dir /b lib\rules > rules.txt` in command prompt, 
removed the `.ts` extension from all rows,
and then appended `"regexp/` and `": "error",` in each row start/end.

&nbsp;
So, is my PR is not helpful? Should I close it? Just asking.

Reply:
No, don't worry, it is helpful. We just have to change `all.ts` a little.

I wouldn't want `all.ts` to be generated like `recommended.ts` because it adds to the complexity of our tooling and isn't necessary.

I was imagining something like this:

```ts
import recommended from "./recommended"
import { rules } from "../utils/rules"

const all: Record<string, string> = {}
for (const rule of rules) {
    all[rule.meta.docs.ruleId] = "error"
}

export = {
    plugins: ["regexp"],
    rules: {
        ...all,
        ...recommended.rules,
    },
}
```

I haven't tested it though.

Reply:
I just committed this, as it seems to be fine.

---

<details>
<summary>Btw, it's maybe a bit off-topic but:</summary>
It's the 2nd PR I made in the repo, 
and I feel once again embarrassed being confused over the same things,
i.e. what README.md should I modify vs which are automatically generated + the usage of `npm run update`.
I feel bad asking for help every time.

So, I'd like to kindly ask for something, please:
could someone make a dedicated CONTRIBUTING.md, clarifying in details: 
- how to make PR for the documentation, 
- how to make PR for the code itself (proposing a new rule, etc).
- how to test the plugin afterwards (I mean how to configure my .eslintrc.js to use the produced `dist` directory)

Regarding 1st and 2nd, I mean, which files to modify, which are to be left intact, 
and don't forget to `npm run update` afterwards.

&nbsp;
The info in README.md is helpful, but not enough:
![2021-06-14_164741](https://user-images.githubusercontent.com/723651/121902797-5437cf00-cd30-11eb-99ae-b4ea1530cd68.jpg)

&nbsp;
I believe it would be very helpful, and facilitate new contributors.

</details>

Reply:
> could someone make a dedicated CONTRIBUTING.md guide

Good point, I'll write one.